### PR TITLE
Improved doc. and example

### LINKED ISSUE #52: Update documentation according to recent changes

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
Closes #52.

I'll comment the changes a bit, where I feel it is appropriate.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,4 +21,38 @@ jobs:
         with:
           otp-version: ${{matrix.otp_vsn}}
           rebar3-version: '3.14'
-      - run: rebar3 test
+      - run: |
+          rm -rf example/_checkouts  # Prevent long running analysis
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/ci.yml:25]: Even though this exception is in `hank`'s config., I'm not certain it's filtering the folders prior to listing them. Local analysis takes a while, but eventually finishes.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,3 @@
-[![Stories in Ready](https://badge.waffle.io/inaka/cowboy-trails.png?label=ready&title=Ready)](https://waffle.io/inaka/cowboy-trails)
```
Comment (paulo-ferraz-oliveira) [inline README.md:1]: This whole file was heavily edited, and is probably best viewed in [Preview mode](https://github.com/inaka/cowboy-trails/blob/a66154c3491a0bd0b49f53741959b1fd5cefdbce/README.md).

File: README.md [hunk 3]
```diff
@@ -9,22 +7,24 @@
 Cowboy routes on steroids!
 
 ## Contact Us
+
 If you find any **bugs** or have a **problem** while using this library, please
-[open an issue](https://github.com/inaka/elvis/issues/new) in this repo
-(or a pull request :)).
+[open an issue](https://github.com/inaka/cowboy-trails/issues/new) in this repo
```
Comment (paulo-ferraz-oliveira) [inline README.md:12]: Now we know who came first 😄 (or where this bit was copied from, too).

File: README.md [hunk 4]
```diff
@@ -128,26 +131,10 @@ Trails =
...
-This way each handler keeps their own routes, as it should be, and you can
+This way each handler maintains their own routes, as it should be, and you can
 merge them easily.
 
 ## Example
 
 For more information about `cowboy_trails`, how to use it and the different
-functions that it exposes, please check this [Example](./example).
-
-## Testing
```
Comment (paulo-ferraz-oliveira) [inline README.md:139]: Meta-testing no longer requires the PLT to be present from the start. At least I cleared `_build` and ran `rebar3 ct` and didn't see any issues.

File: example/README.md [hunk 5]
```diff
@@ -1,72 +1,64 @@
-# Cowboy Trails Example
+# Cowboy Trails Basic Example
```
Comment (paulo-ferraz-oliveira) [inline example/README.md:1]: To keep it close to `.app.src`.

File: example/README.md [hunk 6]
```diff
@@ -1,72 +1,64 @@
...
-   path_match => <<"/message/[:echo]">>},
- #{constraints => [],
-   handler => example_description_handler,
+## Handlers' Description
+
+If you run this `curl` command, you will see all trails' description of this server.
+
+```shell
+$ curl -i http://localhost:8080/description
+[#{constraints => [],handler => example_poor_kv_handler,
```
Comment (paulo-ferraz-oliveira) [inline example/README.md:21]: Copy-pasted from the actual shell output.

File: example/_checkouts/trails [hunk 7]
```diff
@@ -0,0 +1 @@
+../../.
```
Comment (paulo-ferraz-oliveira) [inline example/_checkouts/trails:1]: This'll mean our example will always use the most up to date `trails`.

File: example/rebar.config [hunk 8]
```diff
@@ -4,22 +4,42 @@
 %% == Erlang Compiler ==
```
Comment (paulo-ferraz-oliveira) [inline example/rebar.config:4]: I ran `rebar3 format` over the example. These changes are probably best seen commit by commit.

File: example/rebar.config [hunk 9]
```diff
@@ -4,22 +4,42 @@
...
-           , warn_unused_import
-           , warn_unused_function
-           , warn_bif_clash
-           , warn_unused_record
-           , warn_deprecated_function
-           , warn_obsolete_guard
-           , strict_validation
-           , warn_export_vars
-           , warn_exported_vars
-           , warn_missing_spec
```
Comment (paulo-ferraz-oliveira) [inline example/rebar.config:19]: Removed it.

File: example/rel/sys.config [hunk 10]
```diff
@@ -1,9 +1,4 @@
-[
```
Comment (paulo-ferraz-oliveira) [inline example/rel/sys.config:1]: Was formatted by hand (`format` didn't pick it up). I hope it's Ok.

File: example/src/example.erl [hunk 11]
```diff
@@ -6,48 +6,45 @@
...
-        [
-         {dispatch, Dispatch}
-        ]},
-       {compress, true},
-       {timeout, 12000}
-      ],
-  {ok, _} =
-    cowboy:start_http(example_http, ListenerCount, RanchOptions, CowboyOptions),
-  ok.
+    {ok, Port} = application:get_env(example, http_port),
```
Comment (paulo-ferraz-oliveira) [inline example/src/example.erl:34]: This block was not only formatted but adapted to work with the `cowboy` version `trails` expects.

File: example/src/example_default.erl [hunk 12]
```diff
@@ -1,38 +1,30 @@
 -module(example_default).
```
Comment (paulo-ferraz-oliveira) [inline example/src/example_default.erl:1]: This file was not only formatted but adapted to work with the `cowboy` version `trails` expects.

File: example/src/example_description_handler.erl [hunk 13]
```diff
@@ -1,28 +1,17 @@
 -module(example_description_handler).
```
Comment (paulo-ferraz-oliveira) [inline example/src/example_description_handler.erl:1]: This file was not only formatted but adapted to work with the `cowboy` version `trails` expects.

File: example/src/example_poor_kv_handler.erl [hunk 14]
```diff
@@ -1,67 +1,49 @@
 -module(example_poor_kv_handler).
```
Comment (paulo-ferraz-oliveira) [inline example/src/example_poor_kv_handler.erl:1]: This file was not only formatted but adapted to work with the `cowboy` version `trails` expects.

File: rebar.config [hunk 15]
```diff
@@ -53,7 +53,8 @@
 
 {hank, [
     {ignore, [
-        {"example/**", unnecessary_function_arguments},
```
Comment (paulo-ferraz-oliveira) [inline rebar.config:56]: Some of this was tweaked because `hank` hinted at dead options 😉

File: .github/workflows/ci.yml [hunk 16]
```diff
@@ -21,4 +21,40 @@ jobs:
...
+          R=$(curl -s -w "%{http_code}" "localhost:8080/poor-kv/non-existing-key" -H "content-type: text/plain")
+          [[ $R == "404" ]] || exit 1
+          R=$(curl -s -w "%{http_code}" -XDELETE "localhost:8080/poor-kv/some-key" -H "content-type: text/plain")
+          [[ $R == "204" ]] || exit 1
+          R=$(curl -s -w "%{http_code}" -XDELETE "localhost:8080/poor-kv/some-key" -H "content-type: text/plain")
+          [[ $R == "404" ]] || exit 1
+          _build/default/rel/example/bin/example stop
+      - name: check our example
+        run: |
+          rm -rf _checkouts  # Prevent long running analysis
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/ci.yml:59]: Same note (as the exception mentioned above) applies here.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (elbrujohalcon) [reply .github/workflows/ci.yml:25]: It is not. Hank analyses everything, then filters out the results.

**Review by elbrujohalcon — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (paulo-ferraz-oliveira) [reply .github/workflows/ci.yml:25]: Oh, Ok.

**Comment by paulo-ferraz-oliveira** [comment]
Thanks for merging.

**Pull request merged**